torch_xla/csrc/cross_replica_reduces.cpp

```diff
@@ -11,10 +11,18 @@
 #include "torch_xla/csrc/helpers.h"
 #include "torch_xla/csrc/layout_manager.h"
 #include "torch_xla/csrc/token_handler.h"
+#include "torch_xla/csrc/xla_graph_executor.h"
 
 namespace torch_xla {
 namespace {
 
+struct AllReduceToken {
+  std::shared_ptr<torch::lazy::Value> token;
+  std::mutex lock;
+};
```

Comment:
lol I guess you need to manage the merge conflict with the other pr?

Reply:
Yea, no worries.